### PR TITLE
Contests on frontend

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR is connected to #250

### What's done:
- [x] Added ContestListView - view with all active contests
- [x] Added ContestView - view with info about one given contest
- [x] Added LnkContestProjectController
- [x] Added ContestResult entity in order to get contest scores from backend
- [x] Moved ProjectScoreCard to external functional component
- [x] Added contests button to TopBar
- [x] Added contests button to WelcomeView

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestParticipantsMenu.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        results.forEach { contestResult ->
+            div {
+                className = ClassName("col-10 mb-2")
+                a {
+                    href = "#/${contestResult.organizationName}/${contestResult.projectName}"
+                    className = ClassName("stretched-link")
+                }
+                child(projectScoreCard(), jso {
+                    projectName = "${contestResult.organizationName}/${contestResult.projectName}"
+                    contestScore = contestResult.score.toLong()
```
[the remark this answers is not in the document]
Comment (nulls) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestParticipantsMenu.kt:70]: need to multiply on 100, I guess

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt [hunk 2]
```diff
@@ -538,69 +539,10 @@ class OrganizationView : AbstractView<OrganizationProps, OrganizationViewState>(
...
-            div("col") {
-                h6 {
-                    attrs["style"] = jso<CSSProperties> {
-                        justifyContent = JustifyContent.center
-                        display = Display.flex
-                        alignItems = AlignItems.center
-                    }
-                    +topProject.name
-                }
+            child(projectScoreCard()) {
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/OrganizationView.kt:543]: Ok, will fix it

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ProjectScoreCard.kt [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    var contestScore: Long
+}
+
+/**
+ * Functional component for project score demonstration
+ *
+ * @return ReactElement
+ */
+@Suppress("TOO_LONG_FUNCTION", "LongMethod")
+fun projectScoreCard() = fc<ProjectScoreCardProps> { props ->
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/ProjectScoreCard.kt:38]: Will change `fc` to `FC`

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkContestProjectController.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import org.springframework.web.server.ResponseStatusException
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+/**
+ * Controller for processing links between users and their roles in organizations
+ */
+@RestController
+@RequestMapping("/api/$v1/scores/")
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkContestProjectController.kt:26]: I thought it is too complex, but now it seems ok for me
Will fix it

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestParticipantsMenu.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestParticipantsMenu.kt:70]: It was discussed with @akuleshov7 that we do not need it to be float, but now it seems like I misunderstood something :)

[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestParticipantsMenu.kt]: Well, it depends on what is the output of save-orchestrator
For now I just take the score from `save_cloud.lnk_contest_project` table

**Review by petertrr — APPROVED** [review]

**Pull request merged**